app/scripts/filters/resources.js

```diff
@@ -125,6 +125,52 @@ angular.module('openshiftConsole')
       return displayName;
     };
   })
+  .filter('canI', function(AuthorizationService) {
+    return function(verb, resource, namespace) {
+      return AuthorizationService.canI(resource, verb, namespace);
+    };
+  })
+  .filter('canIAddToProject', function(AuthorizationService) {
+    return function(namespace) {
+      return AuthorizationService.getCanIAddToProject(namespace);
+    };
+  })
+  .filter('canIDoAny', function(canIFilter) {
+    var resourceRulesMap = {
+      "buildConfigs": {"buildconfigs": ["delete", "update"], "buildconfigs/instantiate": ["create"]},
+      "builds": {"builds/clone": ["create"], "builds": ["create", "update"]},
+      "deploymentConfigs": {"horizontalpodautoscalers": ["create", "update"], "deploymentconfigs": ["create", "update"]},
+      "deployments": {"replicationcontrollers": ["update", "delete"]},
+      "horizontalPodAutoscalers": {"horizontalpodautoscalers": ["update", "delete"]},
+      "imageStreams": {"imagestreams": ["update", "delete"]},
+      "persistentVolumeClaims": {"persistentvolumeclaims": ["update", "delete"]},
+      "pods": {"pods": ["update", "delete"], "deploymentconfigs": ["update"]},
+      "replicationControllers": {"horizontalpodautoscalers": ["create", "update"], "replicationcontrollers": ["create", "update"]},
+      "routes": {"routes": ["update", "delete"]},
+      "services": {"services": ["update", "create", "delete"]},
+      "projects": {'projects': ['delete', 'update']}
+    };
+    return function(resource) {
+      var canIDoAny = false;
+      _.each(resourceRulesMap[resource], function(verbs, resource) {
+        _.each(verbs, function(verb) {
+          canIDoAny = canIFilter(resource,verb);
+        });
+      });
+      return canIDoAny;
+    };
+  })
+  .filter('canIScale', function(canIFilter) {
+    return function(deploymentConfig) {
```

Comment:
this should take the actual deployment resource (the RC), we already have an isDeployment filter.  Reduce this function down to:

```
.filter('canIScale', function(canIFilter, isDeploymentFilter) {
  return function(deployment) {
    return canIFilter(isDeploymentFilter(deployment) ? "deploymentconfigs/scale" : "replicationcontrollers", "update"); 
  }
}
```